spec/rubocop/cop/rspec/verified_double_reference_spec.rb

```diff
@@ -0,0 +1,77 @@
+# frozen_string_literal: true
+
+RSpec.shared_examples 'verified double references' do |verified_double|
+  let(:cop_class) { RuboCop::Cop::RSpec::VerifiedDoubleReference }
+
+  context 'when EnforcedStyle is constant' do
+    let(:cop_config) do
+      { 'EnforcedStyle' => 'constant' }
+    end
+
+    it 'does not flag a violation when using a constant reference' do
+      expect_no_offenses("#{verified_double}(ClassName)")
+    end
+
+    it 'flags a violation when using a string reference' do
+      expect_offense(<<~RUBY, verified_double: verified_double)
+        %{verified_double}('ClassName')
+        _{verified_double} ^^^^^^^^^^^ Prefer `constant` references for verified doubles.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        #{verified_double}(ClassName)
+      RUBY
+    end
+
+    include_examples 'detects style',
+                     "#{verified_double}(ClassName)",
+                     'constant'
+  end
+
+  context 'when EnforcedStyle is string' do
+    let(:cop_config) do
+      { 'EnforcedStyle' => 'string' }
+    end
+
+    it 'does not flag a violation when using a string reference' do
+      expect_no_offenses("#{verified_double}('ClassName')")
+    end
+
+    it 'flags a violation when using a constant reference' do
+      expect_offense(<<~RUBY, verified_double: verified_double)
+        %{verified_double}(ClassName)
+        _{verified_double} ^^^^^^^^^ Prefer `string` references for verified doubles.
+      RUBY
+
+      expect_correction(<<~RUBY)
+        #{verified_double}('ClassName')
+      RUBY
+    end
+
+    include_examples 'detects style',
+                     "#{verified_double}('ClassName')",
+                     'string'
+  end
+end
+
+RSpec.describe RuboCop::Cop::RSpec::VerifiedDoubleReference do
+  described_class::RESTRICT_ON_SEND.each do |verified_double|
+    describe verified_double do
+      it_behaves_like 'verified double references', verified_double
```

Comment:
Since this only includes shared examples, and those shared examples are exclusively used from here, I believe it makes sense to just inline those shared examples.

Reply:
Fine with me, inlined